BlueprintUI/Sources/Layout/Alignment.swift

```diff
@@ -0,0 +1,106 @@
+import CoreGraphics
+
+/// Types used to identify alignment guides.
+///
+/// Types conforming to `AlignmentID` have a corresponding alignment guide value,
+/// typically declared as a static constant property of `HorizontalAlignment` or 
+/// `VerticalAlignment`.
+///
+public protocol AlignmentID {
```

Comment:
Does this need to be public? While I think I've wrapped my head around the impl. and why it's needed, I wonder if it's best kept as an implementation detail. Or do you think it's useful to let folks create custom H/V alignments with a custom ID?

Reply:
> Or do you think it's useful to let folks create custom H/V alignments with a custom ID?

I believe this is the goal and aligns w/ how SwiftUI works.

Reply:
It's not very useful today, because these do not propagate, but as Kyle said, this _aligns_ us with SwiftUI. 😏